run_goose.py

```diff
@@ -51,14 +51,29 @@ def main():
     parser.add_argument("--wait", type=int, default=30, help="Wait for connection for X seconds")
     args = parser.parse_args()
 
-    db_connection_string = "postgresql://{}:{}@{}:{}/{}?sslmode=disable".format(
+    
+
+    db_connection_string = "postgresql://{}:{}@{}:{}/{}".format(
         quote(os.environ["MF_METADATA_DB_USER"]),
         quote(os.environ["MF_METADATA_DB_PSWD"]),
         os.environ["MF_METADATA_DB_HOST"],
         os.environ["MF_METADATA_DB_PORT"],
         os.environ["MF_METADATA_DB_NAME"],
     )
 
+    _ssl_mode = os.environ["MF_METADATA_DB_SSL_MODE"]
+    _ssl_cert_path = os.environ["MF_METADATA_DB_SSL_CERT_PATH"]
+    _ssl_key_path = os.environ["MF_METADATA_DB_SSL_KEY_PATH"]
+    _ssl_root_cert_path = os.environ["MF_METADATA_DB_SSL_ROOT_CERT"]
+    ssl_query = ""
+    if(_ssl_mode == "require"):
+        ssl_query = "sslmode=require"
+        if _ssl_cert_path is not None: ssl_query = ssl_query + "&sslcert="+_ssl_cert_path
```

Comment:
Are all these fields optional?

Reply:
Yep all fields are optional, if none are provided then the functionality is the same as before. If MF_METADATA_DB_SSL_MODE = require then the connection string is changed to require ssl. The user can also define MF_METADATA_DB_SSL_CERT_PATH etc. on top of ssl mode = require if they want to specify certificate files or else it will default to systems SSL context.